cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -1,58 +1,223 @@
-require 'fileutils'
-require 'securerandom'
+require 'mixlib/config'
+require 'chef/mixin/deep_merge'
 
-# Manages configuration
-class Supermarket
+module Supermarket
```

Comment:
Well I suppose the inertia of mixlib-config is too hard at this point to overcome, so I'm OK with having my experiment with using the attributes for configuration declared a failure on those grounds.

I think the next step to getting this merged is getting passing ChefSpecs at a minimum.